src/chaat/db/friend_request.clj

```diff
@@ -0,0 +1,60 @@
+(ns chaat.db.friend-request
+  (:require
+   [next.jdbc :as jdbc]
+   [next.jdbc.sql :as sql]
+   [chaat.handler.errors :refer [error-table]]
+   [chaat.db.user :as user]
+   [java-time.api :as jt]))
+
+;; Will be more logical to have the user-exists? and exists? check in the model layer.
+;; Same for the friend request exists? check
+(defn exists?
+  "Check if friend request id exists in friend_requests table"
+  [connection id]
+  (not (empty? (sql/find-by-keys connection :friend_requests {:id id}))))
+
+(defn insert
+  "Insert a friend request into the db"
+  [db {:keys [sender_username recipient_username] :as content}]
+  (jdbc/with-transaction [tx (db)]
+    (if (and (user/user-exists? tx sender_username)
+             (user/user-exists? tx recipient_username))
+      (let [query-result (sql/insert! tx :friend_requests content)]
```

Comment:
Need to handle what happens when the insert fails.

Reply:
The let block might be superfluous if we're not handling those errors. We could just inline `(sql/insert! tx :friend_requests content)`